lib/config/fileconf.go

```diff
@@ -1300,6 +1305,67 @@ func (dt *DeviceTrust) Parse() (*types.DeviceTrust, error) {
 	}, nil
 }
 
+// HostedPlugins defines 'auth_service/plugins' Enterprise extension
+type HostedPlugins struct {
+	Enabled        bool                 `yaml:"enabled"`
+	OAuthProviders PluginOAuthProviders `yaml:"oauth_providers,omitempty"`
+}
+
+// PluginOAuthProviders holds application credentials for each
+// 3rd party API provider.
+type PluginOAuthProviders struct {
+	Slack *OAuthClientCredentials `yaml:"slack,omitempty"`
+}
+
+func (p *PluginOAuthProviders) Parse() (service.PluginOAuthProviders, error) {
+	out := service.PluginOAuthProviders{}
+	if p.Slack == nil {
+		return out, trace.BadParameter("when plugin runtime is enabled, at least one plugin provider must be specified")
+	}
+
+	slack, err := p.Slack.Parse()
+	if err != nil {
+		return out, trace.Wrap(err)
+	}
+	out.Slack = slack
+	return out, nil
+}
+
+// OAuthClientCredentials hold client credentials for Teleport's OAuth app.
+type OAuthClientCredentials struct {
+	ClientID     string `yaml:"client_id"`
+	ClientSecret string `yaml:"client_secret"`
+}
+
+func (o *OAuthClientCredentials) Parse() (*oauth2.ClientCredentials, error) {
+	if o.ClientID == "" || o.ClientSecret == "" {
+		return nil, trace.BadParameter("Both client_id and client_secret must be specified")
+	}
+	clientID := o.ClientID
+	clientSecret := o.ClientSecret
+
+	// If the values look like filepaths, try to read from the paths
+	if strings.HasPrefix(clientID, "/") {
+		content, err := os.ReadFile(clientID)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		clientID = strings.TrimSpace(string(content))
+	}
+	if strings.HasPrefix(clientSecret, "/") {
```

Comment:
Should we disallows anything that's not a path? I think our approach has been to now allow embedding secrets in the configs and only support reading from files.